pkgs/development/cuda-modules/cudatoolkit/default.nix

```
@@ -292,15 +295,24 @@ backendStdenv.mkDerivation rec {
         wrapProgram $out/bin/nvprof \
           --prefix LD_LIBRARY_PATH : $out/lib
       ''
-    # 11.8 includes a broken symlink, include/include, pointing to targets/x86_64-linux/include
-    + lib.optionalString (lib.versions.majorMinor version == "11.8") ''
+    # 11.8 and 12.9 include a broken symlink, include/include, pointing to targets/x86_64-linux/include
+    + lib.optionalString (lib.versions.majorMinor version == "11.8" || lib.versions.majorMinor version == "12.9") ''
       rm $out/include/include
     ''
+    # 12.9 has another broken symlink, lib64/lib64, pointing to lib/targets/x86_64-linux/lib
+    + lib.optionalString (lib.versions.majorMinor version == "12.9") ''
+      rm $out/lib64/lib64
+    ''
     # Python 3.8 is not in nixpkgs anymore, delete Python 3.8 cuda-gdb support
     # to avoid autopatchelf failing to find libpython3.8.so.
     + lib.optionalString (lib.versionAtLeast version "12.6") ''
       find $out -name '*python3.8*' -delete
     ''
+    # Python 3.9 is not in nixpkgs anymore, delete Python 3.9 cuda-gdb support
+    # to avoid autopatchelf failing to find libpython3.9.so.
+    + lib.optionalString (lib.versionAtLeast version "12.9") ''
+      find $out -name '*python3.9*' -delete
+    ''
```

I think this is the correct way how to do it.

If python3.9 is not available anymore, then it surely affects older CUDA versions too.

See also related commit bc27118dc756edc5e2633f7a787878ae98b6b4cb which is already on master - I wonder why this commit uses `versionAtLeast 12.5` while cudatoolkit uses `versionAtLeast 12.6` - can you investigate also whether this is fine? Or one of the versions is wrong and we should have the same in both?

```suggestion
    # Python 3.8 and 3.9 are not in nixpkgs anymore, delete Python 3.{8,9} cuda-gdb support
    # to avoid autopatchelf failing to find libpython3.{8,9}.so.
    + lib.optionalString (lib.versionAtLeast version "12.6") ''
      find $out -name '*python3.8*' -delete
      find $out -name '*python3.9*' -delete
    ''
```

---

Fixed.